### PR TITLE
Update logos and text references to BCFP

### PR DESCRIPTION
**Pull request by Scotchester**
Only thing done so far is the main header, behind a new `BCFP_LOGO` feature flag.

## Additions

- BCFP logos and homepage hero images
- `BCFP_LOGO` feature flag

## Removals

- Unused `home-hero_med.jpg` image
- Unused Ask CFPB `_search.html` template
- Unused Find a Housing Counselor PDF view (this is not what is currently used to generate our PDFs)
- Unused `base_print.html` legacy template

## Changes

- If `BCFP_LOGO` flag is on:
  - Display BCFP logo in main header (including on Spanish site)
  - Update text of homepage hero
  - Update page titles
  - Display BCFP logo on print view in Find a Housing Counselor and updates the name in the text on that view
  - Display BCFP logo as default Open Graph image (sharing to Facebook and Twitter) when a page-specific one is not given

## Testing

1. Pull branch
1. `gulp images`
1. Turn the `BCFP_LOGO` flag on
1. Load up http://localhost:8000/ and see the new logo, hero text and graphic
1. View it in responsive design mode and make sure all widths look good
1. Go to http://localhost:8000/es/ and see the new logo
1. Go to http://localhost:8000/find-a-housing-counselor/ and Print Preview and see the new logo
1. Back on the homepage, View Source and see that it's specifying the new `bfcp-logo` Open Graph images (the `<meta property="og:image" …>` tags

## Todos

- Replace other instances listed in GHE#2872
- Clean up Feature Flag once it's been flipped on in production

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
- [x] Passes all existing automated tests
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future todos are captured in comments
- [x] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [x] Project documentation has been updated
- [x] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Browsers

- [x] Chrome
- [x] Firefox
- [x] Safari
- [ ] Internet Explorer 8, 9, 10, and 11
- [ ] Edge
- [ ] iOS Safari
- [ ] Chrome for Android

### Accessibility

- [x] Keyboard friendly
- [x] Screen reader friendly

### Other

- [x] Is useable without CSS
- [x] Is useable without JS
- [x] Flexible from small to large screens
- [x] No linting errors or warnings
- [x] JavaScript tests are passing

### CODE REVIEW

File: cfgov/jinja2/v1/_includes/organisms/header.html [hunk 1]
```diff
@@ -46,6 +46,19 @@
             {% include 'molecules/global-search.html' with context %}
 
             <a class="o-header_logo" href="/">
+            {% if flag_enabled('BCFP_LOGO', request) %}
+                <img class="o-header_logo-img"
```
Comment (jimmynotjim) [inline cfgov/jinja2/v1/_includes/organisms/header.html]: Is it worth wrapping just the src, srcset, and sizes and leaving the tags, class, and alt outside the flag?

Comment (Scotchester) [reply cfgov/jinja2/v1/_includes/organisms/header.html]: Well, the `alt` has to change, too... it seemed cleaner to just wrap the whole tag.

Comment (jimmynotjim) [reply cfgov/jinja2/v1/_includes/organisms/header.html]: Oh, right, I've already become blind to the change...carry on.

**Comment by Scotchester** [comment]
This has been rebased on master and force-pushed, and is now ready for final review.

**Comment by schaferjh** [comment]
Just checking -- if `BCFP_LOGO` flag is on: the homepage hero image is update as well as the text, right?

**Comment by Scotchester** [comment]
@schaferjh That's correct. That work is in commit 40c0e90.

**Pull request merged**